### PR TITLE
Adds python 3.8 support

### LINKED ISSUE #16: Support Python 3.8
Need to build binaries for all platforms

### PR DESCRIPTION
**Pull request by vmoeykens**
Closes #16

### CODE REVIEW

**Comment by jauerb** [comment]
@cullen-jemison @vmoeykens when I tried this right now I get 

```
ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/josh/anaconda3/envs/flight-data-visualizer-py38/lib/python3.8/site-packages/nanogui.cpython-38-x86_64-linux-gnu.so)
```

also there a reason we say <= 3.8.2 rather than allowing all 3.8 versions?  Think that should be changed to < 3.9 unless there is a reason things will break > 3.8.2

**Comment by jauerb** [comment]
> @cullen-jemison @vmoeykens when I tried this right now I get
> 
> ```
> ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/josh/anaconda3/envs/flight-data-visualizer-py38/lib/python3.8/site-packages/nanogui.cpython-38-x86_64-linux-gnu.so)
> ```
> 
> also there a reason we say <= 3.8.2 rather than allowing all 3.8 versions? Think that should be changed to < 3.9 unless there is a reason things will break > 3.8.2

OK, got it working but had to 
```
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt update
sudo apt upgrade
```
to get it working on Ubuntu 18.04

**Comment by jauerb** [comment]
Not ideal for user to have to upgrade their C++ version to be able to use this.

**Comment by cullen-jemison** [comment]
I actually was able to build the Linux binary last night. It's in the bin folder. I requested that @cwoodall build the Mac binary, but we'll still need the Windows one.

**Comment by jauerb** [comment]
I reali

> I actually was able to build the Linux binary last night. It's in the bin folder. I requested that @cwoodall build the Mac binary, but we'll still need the Windows one.

Yes, this is what I was trying to use when I pip installed this branch on Linux, but ran into issue described above.

**Comment by cullen-jemison** [comment]
Oh interesting, I thought you were trying to build it. Was it trying to build a wheel?

**Comment by jauerb** [comment]
> Oh interesting, I thought you were trying to build it. Was it trying to build a wheel?

It installed fine, this happened at runtime.  Apparently linked to a newer version of C++ than I had, so maybe just an issue with how you built it.

File: setup.py [hunk 1]
```diff
@@ -35,17 +51,20 @@
...
     package_data = os_distributions[sys.version_info[1]]
 except KeyError:
     # don't use f-strings to prevent syntax error on older Python versions
-    sys.exit('Currently we only support Python 3.6 or 3.7, but you are using ' + sys.version)
+    sys.exit('Currently we only support Python 3.6, 3.7 or 3.8, but you are using ' + sys.version)
 
 setup(
     name='nanogui',
-    python_requires='>=3.6, <3.8',
+    python_requires='>=3.6, <=3.8.2',
```
Comment (jauerb) [inline setup.py:67]: why 3.8.2?

**Pull request merged**